src/crm/crm.controller.ts

```diff
@@ -77,15 +86,22 @@ export class CrmController {
 
   @Get('refresh-companies')
   @Roles(UserRole.admin)
-  async refreshCompanies(@Query('campaignId') campaignId: number) {
+  async refreshCompanies(@Query() { campaignId }: RefreshCompanySchema) {
     return await this.crmCampaignsService.refreshCompanies(campaignId)
   }
 
+  @Get('mass-refresh-companies')
+  @Roles(UserRole.admin)
+  async massRefreshCompanies(@Query() { fields }: MassRefreshCompanySchema) {
+    return await this.crmCampaignsService.massRefreshCompanies(
+      fields as Array<keyof CRMCompanyProperties>,
```

Comment:
We wouldn't need to do this type-casting if we use the keys from `CRMCompanyProperties` for the Zod `MassRefreshCompanySchema.fields` instead of just `z.array(z.string())`, right?